compiler/noirc_frontend/src/hir_def/types.rs

```diff
@@ -1587,6 +1588,9 @@ impl Type {
         }
     }
 
+    /// Check whether this type is itself a slice, or a struct/enum/tuple which contains a slice in a field or variant.
+    ///
+    /// Note that arrays cannot contain slices, so for them it always returns `false`.
```

Comment:
Thanks to this comment I realized there's a minor bug here. This doesn't give an error:

```noir
pub struct Foo {
    x: [[[Field]; 3]; 3],
}

fn main () {}
```

but it probably should as there's a nested slice (but we don't look into an array of an array).

We still get an error if we try to create `Foo`, but during monomorphization I think.